code/datums/action.dm

```diff
@@ -0,0 +1,357 @@
+	name = "Toggle Gunlight"
+
+/datum/action/item_action/toggle_mode
+	name = "Toggle Mode"
+
+/datum/action/item_action/toggle_barrier_spread
+	name = "Toggle Barrier Spread"
+
+/datum/action/item_action/equip_unequip_TED_Gun
+	name = "Equip/Unequip TED Gun"
+
+/datum/action/item_action/toggle_paddles
+	name = "Toggle Paddles"
+
+/datum/action/item_action/set_internals
+	name = "Set Internals"
+
+/datum/action/item_action/set_internals/UpdateButtonIcon()
+	if(..()) //button available
+		if(iscarbon(owner))
+			var/mob/living/carbon/C = owner
+			if(target == C.internal)
+				button.icon_state = "bg_internals_on"
+
+/datum/action/item_action/toggle_mister
+	name = "Toggle Mister"
+
+/datum/action/item_action/activate_injector
+	name = "Activate Injector"
+
+/datum/action/item_action/toggle_helmet_light
+	name = "Toggle Helmet Light"
+
+/datum/action/item_action/toggle_helmet_flashlight
+	name = "Toggle Helmet Flashlight"
+
+/datum/action/item_action/toggle_helmet_mode
+	name = "Toggle Helmet Mode"
+
+/datum/action/item_action/toggle
+
+/datum/action/item_action/toggle/New(Target)
+	..()
+	name = "Toggle [target.name]"
+	button.name = name
+
+/datum/action/item_action/halt
+	name = "HALT!"
+
+/datum/action/item_action/toggle_voice_box
+	name = "Toggle Voice Box"
+
+/datum/action/item_action/change
+	name = "Change"
+
+/datum/action/item_action/adjust
+
+/datum/action/item_action/adjust/New(Target)
+	..()
+	name = "Adjust [target.name]"
+	button.name = name
+
+/datum/action/item_action/switch_hud
+	name = "Switch HUD"
+
+/datum/action/item_action/toggle_wings
+	name = "Toggle Wings"
+
+/datum/action/item_action/toggle_human_head
+	name = "Toggle Human Head"
+
+/datum/action/item_action/toggle_helmet
+	name = "Toggle Helmet"
+
+/datum/action/item_action/jetpack_mode
+	name = "Jetpack Mode"
+
+/datum/action/item_action/hands_free
+	check_flags = AB_CHECK_CONSCIOUS
+
+/datum/action/item_action/hands_free/activate
+	name = "Activate"
+
+
+/datum/action/item_action/hands_free/shift_nerves
+	name = "Shift Nerves"
+
+
+/datum/action/item_action/toggle_research_scanner
+	name = "Toggle Research Scanner"
+	button_icon_state = "scan_mode"
+
+/datum/action/item_action/toggle_research_scanner/Trigger()
+	if(IsAvailable())
+		owner.research_scanner = !owner.research_scanner
+		owner << "<span class='notice'>Research analyzer is now [owner.research_scanner ? "active" : "deactivated"].</span>"
+		return 1
+
+/datum/action/item_action/toggle_research_scanner/Remove(mob/living/L)
+	if(owner)
+		owner.research_scanner = 0
+	..()
+
+/datum/action/item_action/toggle_research_scanner/ApplyIcon(obj/screen/movable/action_button/current_button)
+	if(button_icon && button_icon_state)
+		var/image/img = image(button_icon, current_button, "scan_mode")
+		current_button.overlays += img
+
+/datum/action/item_action/organ_action
+	check_flags = AB_CHECK_CONSCIOUS
+
+/datum/action/item_action/organ_action/New(Target)
+	..()
+	name = "Toggle [target.name]"
```

Comment:
Because the current button names used by organ actions are all pretty much "Toggle [theorganname]" , check the old "organ_action_name = "stuff" "